app_dart/test/request_handlers/github_webhook_test.dart

```diff
@@ -2037,6 +2041,73 @@ void foo() {
     });
   });
 
+  group('github webhook create branch event', () {
+    test('should add branch to db if db is empty', () async {
```

Comment:
Most of these tests should be moved into the test for the branch service. This should verify that branch service is called only for the create event

Reply:
Sure, I moved the main testing logics to branch_service_test. 

It is kind of hard to verify branchService is called for create event in github_webhook_test for the following reasons: 
1. verify calls need to be called on a mockito object, and if we create a mock branchService object in test, it is a different instance of branchService from the one created in github_webhook.dart.
2. if we add branch service as a field to github_webhook, since the class is immutable we have to make branch service a final variable and provide it with a const default value. The final field constraint would prevent us from creating a branchService later in the code. And we don't have a good default const constructor for branchService because it relies on a datastore service.
3. If we mark the key function in branch service as visible for testing, the function will be constrained to the class and tests, and it can't be called in github_webhook.
4. If we create a static handling function in github_webhook.dart, we will need a mock instance of github_webhook, which doesn't seem ideal on top of the already non ideal static function.

I ended up creating a default error case for, the switch statement on event types pullrequest, checkrun and create. Therefore if we encounter an event type not conforming to the fore mentioned three types, the program will throw an error. And for github_webhook test, I made create event test similar to checkrun test in that, it won't check for how many times the function is called. Instead it only checks if the test runs without causing an exception. And we trust the switch statement categorizes event types correctly without adding a test to it.

Reply:
Throwing an error on the webhooks will cause a large number of errors in our backend.

1 is the right idea, and you in the case no branch service was passed, it can be constructed like the previously logic had.

Reply:
Thanks for the reply! umm does `in the case no branch service was passed` imply that we are passing branchService as a field variable to github_webhook? I think the challenge is that github_webhook is an immutable class which requires the branchService passed in to be final. If that is the case, we would need something like `GithubWebhook(branchService: BranchService(DatastoreService(config.db, maxentitygroup), requestString)` when we are initializing GithubWebhook, but at that point we can't properly provide the information of maxentitygroup and requestString?

Reply:
Correct. Feel free to remove the immutability status of the handlers. For server handlers, ti doesn't matter much, but it's helpful for models

Reply:
Thanks for the explanations! I removed the immutability of request handler class to better serve tests